examples/ProtoBuffer/Receive/Receive.ino

```diff
@@ -0,0 +1,91 @@
+#include <TheThingsNetwork.h>
+#include <TheThingsMessage.h>
+
+// Set your AppEUI and AppKey
+const char *appEui = "0000000000000000";
+const char *appKey = "00000000000000000000000000000000";
+
+#define loraSerial Serial1
+#define debugSerial Serial
+
+TheThingsNetwork ttn(loraSerial, debugSerial, /* TTN_FP_EU868 or TTN_FP_US915 */);
+
+sensordata_t sensorData = api_SensorData_init_default;
+getdata_t getData = api_GetData_init_default;
+
+void setup()
+{
+  pinMode(2, INPUT);
+
+  loraSerial.begin(57600);
+  debugSerial.begin(9600);
+
+  // Wait a maximum of 10s for Serial Monitor
+  while (!debugSerial && millis() < 10000);
+
+  debugSerial.println("-- STATUS");
+  ttn.showStatus();
+
+  debugSerial.println("-- JOIN");
+  ttn.join(appEui, appKey);
+
+  ttn.onMessage(message);
+
+  sensorData.has_motion = false;
+  sensorData.has_water = false;
+  sensorData.has_temperature_celcius = false;
+  sensorData.has_temperature_fahrenheit = false;
+  sensorData.has_humidity = false;
+
+  getData.customdata.has_foo = true;
+}
+
+void loop() {
+  debugSerial.println("-- LOOP");
+
+  // Read sensors
+  bool motion = 0;
+  uint32_t water = 682 - analogRead(A0);
+  uint32_t foo = 13;
+
+  // Init data
+  sensorData.motion = motion;
+  sensorData.water = water;
+  getData.customdata.foo = foo;
+
+  // Encode data
+  byte *buffer;
+  size_t size;
+  int8_t port = 1;
+
+  // Send standard message on port 100, if every sensordata message are false then you send a custom message on a different port (1 by default)
+  if (TheThingsMessage::encodeSensorData(&sensorData, &buffer, &size, &port) == SEND_CUSTOM) {
+    TheThingsMessage::encodeCustomData(&getData.customdata, &buffer, &size);
+  }
+
+  ttn.sendBytes(buffer, size, port);
+
+  delay(10000);
+}
+
+void message(const byte* payload, size_t length, port_t port) {
+  if (port >= 100) {
+    //standard message always received on port 100 or more
+    getData.appdata = api_AppData_init_default;
+    getData.has_appdata = true;
+  } else {
+    //otherwise we get a custom message
+    getData.customdata = api_CustomData_init_default;
+    getData.has_customdata = true;
+  }
+
+  TheThingsMessage::decodeGetData(&getData, payload, length);
```

Comment:
Just like we have a `encodeSensorData`, we have a `decodeAppData`. Custom messaging requires a custom decoder and passing fields. Maybe that's beyond the scope of this example here. Keep it simple